src/cpu/flexcpu/simple_dataflow_thread.cc

```diff
@@ -757,6 +870,22 @@ SDCPUThread::populateDependencies(shared_ptr<InflightInst> inst_ptr)
 
     // END ISA explicit serialization
 
+    if (static_inst->isNonSpeculative()) {
+        const shared_ptr<InflightInst> last_inst = *(++inflightInsts.rbegin());
```

Comment:
This could be a function, too, for the same reason as below (e.g., `getYoungestInst()`). Also, it would be best to use youngest/oldest instead of first/last.

Reply:
I think both of these comments might be solved by a function along the lines of "get the youngest instruction that is still yet to be committed, but is still alive on the buffer"